### ISSUE TITLE
Allow loop argument for initializer() and finalizer() function

### ISSUE THREAD

**Issue by jrmi** [tortoise/tortoise-orm#57]
While testing my code using tortoise-orm, I created a specific `event_loop` and dropped the default thread event loop. As consequence the method `get_event_loop()` is not working as excepted. 
The problem is that `initializer` and `finalizer` methods use the asyncio method to get the event loop and there is no other way.

I would like to be able to give the event loop as an optional argument with default value to what we get with the `get_event_loop()` method.

**Comment by grigi** [comment]
What test runner are you using?
I have very recently revamped the test utils to mess around less with the event loop. Might be useful to take that + fixes for you together.
Could you post a small example of how you did it so I can use that as a test case?

**Comment by jrmi** [comment]
I'm trying to use pytest which seems not to be the recommended tool with tortoise ;-)  but seems working. I can't use the contrib TestCase so i tried to simulate the same feature with pytest but the loop argument was missing.

Here the code to reproduce: 

```python
import asyncio
from tortoise.contrib.test import initializer

asyncio.set_event_loop(None)
loop = asyncio.new_event_loop()

initializer() # Should be initializer(loop=loop)
```

**Comment by grigi** [comment]
We have a policy to try and minimise Tortoise dictate project structure, and I think dictating test runner is part of it. So happy to make the unittest runner be easy to integrate with pytest, and nose2.
(I actually had this thought to add it to a project timeline.)

So, I'll take the existing test runner refactor, make it work with a custom event loop as a first step.

**Comment by Zeliboba5** [comment]
@jrmi While I hope we will make tortoise pytest-friendly soon enough - here is example of fixtures for tortoise tests, that I am using in my project, it creates db on test session start and runs each test in separate transaction (there is also pytest-aiohttp plugin used for `loop` fixture and async tests)

```
async def init_tortoise(config):
    try:
        await Tortoise.init(config)
        rules_connection = Tortoise.get_connection('default')
        await rules_connection.close()
        await rules_connection.db_delete()
    except DBConnectionError as exc:
        pass

    rules_connection = AsyncpgDBClient(connection_name='default', **settings.RULES_DB_CONFIG)
    await rules_connection.db_create()
    await Tortoise.init(config)
    await Tortoise._reset_connections()

@pytest.fixture(scope='session')
def tortoise_config(request):
    loop = get_event_loop()
    config = <your actual test config looking at separate test db that can be deleted>
    loop.run_until_complete(init_tortoise(config))
    loop.run_until_complete(run_migrations(db_config=settings.RULES_DB_CONFIG))
    return config

def _swap_connection(connection):
    old_connection = Tortoise.get_connection('default')
    Tortoise._connections['default'] = connection
    return old_connection

@pytest.fixture
async def app(loop, tortoise_config, request):
    # some app initialising here
    await Tortoise.init(tortoise_config)

    transaction = await start_transaction('default')
    connection = _swap_connection(transaction)

    def fin():
        _set_connection(connection)
        loop.run_until_complete(transaction.rollback())
        loop.run_until_complete(Tortoise._reset_connections())

    request.addfinalizer(fin)
    return app
```

It's looks a little bit hacky, but still does the work

**Comment by grigi** [comment]
Ooh, I wanted to point to #59 (which is failing due to GH outage, I hope it gets back soon)(branch: `feature/testrunner_overhaul`) that you can try to see if it fixes your case?
I back ported the testrunner fixes I did to allow `:memory:` databases to work, and allowed `initializer(loop=loop)` to work.

As someone that has been firmly in the UnitTest part of the testing scene, could you help me do this right?
I was planning to write a  py.test [plugin hook](https://docs.pytest.org/en/latest/writing_plugins.html#writing-hook-functions) to start with, that should just call initializer and finalizer.
Will need help to understand how the function-based tests on py.test work, so I don't break it for you.

Currently the testrunner affects global state, which is also not ideal, and to get things to actually work as expected we need to have all tests run in the same event loop, which is also not ideal.

**Comment by grigi** [comment]
Ooh, I wanted to point to #59 (which is failing due to GH outage, I hope it gets back soon)(branch: `feature/testrunner_overhaul`) that you can try to see if it fixes your case?
I back ported the testrunner fixes I did to allow `:memory:` databases to work, and allowed `initializer(loop=loop)` to work.

As someone that has been firmly in the UnitTest part of the testing scene, could you help me do this right?
I was planning to write a  py.test [plugin hook](https://docs.pytest.org/en/latest/writing_plugins.html#writing-hook-functions) to start with, that should just call initializer and finalizer.
Will need help to understand how the function-based tests on py.test work, so I don't break it for you.

Currently the testrunner affects global state, which is also not ideal, and to get things to actually work as expected we need to have all tests run in the same event loop, which is also not ideal.

**Comment by jrmi** [comment]
Thanks @Zeliboba5 i'll try to use this.

@grigi I'll test #59 and keep you informed of the result.

Also i'll try to figure out what I want with py.test. For now I was more thinking about something like a [fixture with a yield inside](https://docs.pytest.org/en/3.9.2/fixture.html#fixture-finalization-executing-teardown-code) to call inializer and finalizer function.

**Comment by grigi** [comment]
Will be fixed for v0.10.10

**Comment by grigi** [comment]
@jrmi Tortoise ORM 0.10.10 is out on PyPi, can you please test that it resolved this and #58 #60 #63

**Comment by jrmi** [comment]
I've just tested all problematic use cases with the new version and seems every fixed things works for me. Thanks @grigi !
